langchain-core/src/structured_query/tests/functional.test.ts

```diff
@@ -0,0 +1,310 @@
+          new Document({
+            pageContent: "",
+            metadata: {
+              stringValue: "val",
+              numberValue: 0,
+              booleanValue: true,
+            },
+          }),
+        ],
+        [Comparators.lte]: [
+          // test for less than
+          new Document({
+            pageContent: "",
+            metadata: {
+              stringValue: "val",
+              numberValue: 0,
+              booleanValue: true,
+            },
+          }),
+          // test for equal to
+          new Document({
+            pageContent: "",
+            metadata: {
+              stringValue: "value",
+              numberValue: 1,
+              booleanValue: true,
+            },
+          }),
+        ],
+      };
+      
+      // documents that will not match against the comparison
+      const invalidDocumentsByComparator: { [key in string]: Document<Record<string, unknown>>[] } = {
+        [Comparators.eq]: [
+          new Document({
+            pageContent: "",
+            metadata: {
+              stringValue: "not-value",
+              numberValue: 0,
+              booleanValue: false,
+            },
+          }),
+        ],
+        [Comparators.ne]: [
+          new Document({
+            pageContent: "",
+            metadata: {
+              stringValue: "value",
+              numberValue: 1,
+              booleanValue: true,
+            },
+          }),
+        ],
+        [Comparators.gt]: [
+          new Document({
+            pageContent: "",
+            metadata: {
+              stringValue: "value",
+              numberValue: 1,
+              booleanValue: true,
+            },
+          }),
+        ],
+        [Comparators.gte]: [
+          new Document({
+            pageContent: "",
+            metadata: {
+              stringValue: "val",
+              numberValue: 0,
+              booleanValue: false,
+            },
+          }),
+        ],
+        [Comparators.lt]: [
+          new Document({
+            pageContent: "",
+            metadata: {
+              stringValue: "valueee",
+              numberValue: 2,
+              booleanValue: true,
+            },
+          }),
+        ],
+        [Comparators.lte]: [
+          new Document({
+            pageContent: "",
+            metadata: {
+              stringValue: "valueee",
+              numberValue: 2,
+              booleanValue: true,
+            },
+          }),
+        ],
+      };
+      
+      const stringComparators = translator.getAllowedComparatorsForType("string");
+      for (const comparator of stringComparators) {
+        const attribute = attributesByType.string;
+        const value = inputValuesByAttribute[attribute];
+        const validDocuments = validDocumentsByComparator[comparator];
+        for (const validDocument of validDocuments) {
+          test(`${value} -> ${comparator} -> ${validDocument.metadata[attribute]}`, () => {
```

Comment:
When a test fails, the generated test name looks like `input_value -> comparator -> document_value`, like shown in the below screenshot for a failed `eq` comparator check, where the input value is `value` and the document value is `invalid-value`.

![image](https://github.com/user-attachments/assets/80abfb18-bccf-4825-9b84-93d5dc1864e8)